### PR TITLE
Allow identities with optouts to re-register

### PR DESCRIPTION
**Pull request by KaitCrawford**
Currently mothers who have opted out of communications will not be able to re-register on the public line because they will be directed to the change menu state.
This change instead retrieves the information we already have for them and sends them straight to the period month state.

### CODE REVIEW

**Comment by KaitCrawford** [comment]
@rudigiesler ready for review. I'm just not sure where the best place to remove the optout for the phone number is, let me know if you have any ideas please.

File: test/ussd_public.test.js [hunk 1]
```diff
@@ -271,6 +271,32 @@ describe("familyconnect health worker app", function() {
...
+                            "What month did you start your last period?",
+                            "1. Apr 15",
+                            "2. Mar 15",
+                            "3. Feb 15",
+                            "4. Jan 15",
+                            "5. Dec 14",
+                            "6. Nov 14",
+                            "7. Oct 14",
+                            "8. Sep 14",
+                            "9. Aug 14"
```
Comment (rudigiesler) [inline test/ussd_public.test.js]: This date format has changed in the base branch, which is why you're getting test errors.

**Review by rudigiesler — COMMENTED** [review]
Maybe the best place to remove the optout from the identity is on a successful registration? So if the contact has optedout=True, then we update the contact to optedout=False as part of the registration creation?

**Comment by KaitCrawford** [comment]
@rudigiesler yeah, I decided to set it in `set_public_mother_details` which is only ever called from `finish_public_registration`

**Review by rudigiesler — APPROVED** [review]
:+1:

**Pull request merged**